Cql/Elm/ElmTreeWalker.cs

```diff
@@ -0,0 +1,92 @@
+/* 
+ * Copyright (c) 2023, NCQA and contributors
+ * See the file CONTRIBUTORS for details.
+ * 
+ * This file is licensed under the BSD 3-Clause license
+ * available at https://raw.githubusercontent.com/FirelyTeam/firely-cql-sdk/main/LICENSE
+ */
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+
+namespace Hl7.Cql.Elm
+{
+    /// <summary>
+    /// A walker that does a depth-first traversal of a tree and calls a visitor function for each node.
+    /// </summary>
+    public class ElmTreeWalker
```

Comment:
Not sure we should make this public.  It's useful for others, that's true. What do we think?

Reply:
Let's keep things internal, until we review the public api later?